packages/lit-dev-content/site/docs/v3/data/task.md

```diff
@@ -0,0 +1,440 @@
+        initial: () => html`<p>Waiting to start task</p>`,
+        pending: () => html`<p>Running task...</p>`,
+        complete: (value) => html`<p>The task completed with: ${value}</p>`,
+        error: (error) => html`<p>Oops, something went wrong: ${error}</p>`,
+      })}
+    `;
+  }
+```
+
+### Running tasks
+
+By default, Tasks will run any time the arguments change. This is controlled by the `autoRun` option, which defaults to `true`.
+
+#### Auto-run
+
+In auto-run mode, the task will call the `args` function when the host has updated, compare the args to the previous args, and invoke the task function if they have changed.
+
+#### Manual mode
+
+If `autoRun` is set to false, the task will be in _manual_ mode. In manual mode you can run the task by calling the `.run()` method, possibly from an event handler:
+
+{% switchable-sample %}
+
+```ts
+class MyElement extends LitElement {
+
+  private _getDataTask = new Task(
+    this,
+    {
+      task: async () =>
+        const response = await fetch(`example.com/data/`);
+        return response.json();
+      ),
+      args: () => []
+    }
+  );
+
+  render() {
+    return html`
+      <button @click=${this._onClick}>Get Data</button>
+    `;
+  }
+
+  private _onClick() {
+    this._getDataTask.run();
+  }
+}
+```
+
+```js
+class MyElement extends LitElement {
+
+  _getDataTask = new Task(
+    this,
+    {
+      task: async () =>
+        const response = await fetch(`example.com/data/`);
+        return response.json();
+      ),
+      args: () => []
+    }
+  );
+
+  render() {
+    return html`
+      <button @click=${this._onClick}>Get Data</button>
+    `;
+  }
+
+  _onClick() {
+    this._getDataTask.run();
+  }
+}
+```
+
+{% endswitchable-sample %}
+
+In manual mode you can provide new arguments directly to `run()`:
+
+```ts
+this._task.run('arg1', 'arg2');
+```
+
+If arguments are not provided to `run()`, they are gathered from the `args` callback.
+
+### Aborting tasks
+
+The second argument to the task function is an options object that carries an [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) in the `signal` property. This can be used to cancel a pending task because a new task run has started.
+
+The easiest way to use a signal is to forward it to an API that accepts it, like `fetch()`.
+
+{% switchable-sample %}
+
+```ts
+  private _task = new Task(this, {
+    task: async (args, {signal}) => {
+      const response = await fetch(someUrl, {signal});
+      // ...
+    },
+  });
+```
+
+```js
+  _task = new Task(this, {
+    task: async (args, {signal}) => {
+      const response = await fetch(someUrl, {signal});
+      // ...
+    },
+  });
+```
+
+{% endswitchable-sample %}
+
+Forwarding the signal to `fetch()` will cause the browser to cancel the network request if the signal is aborted.
+
+You can also check if a signal has been aborted in your task function. It's often useful to race a Promise against a signal:
```

Comment:
This says "race a signal against a promise", but then the example doesn't look like it's racing. I'd expect some kind of `Promise.race` usage.

Reply:
Promise.race is a way to race multiple promises against each other, and it's not the only way to race promises. In this case we're not racing multiple promises though, but an abort signal and promises.

Reply:
fixed